artiq/frontend/artiq_coremgmt.py

```diff
@@ -201,7 +204,92 @@ def main():
     if args.tool == "debug":
         if args.action == "allocator":
             mgmt.debug_allocator()
-
+    
+    if args.tool == "flash":
+
+        bin_dir = args.dir
+        if bin_dir is None:
+            bin_dir = os.path.join(artiq_dir, "board-support")
+        needs_artifacts = not args.partition or any(
+            partition in args.partition
+            for partition in ["gateware", "bootloader", "firmware"])
+        variant = args.variant
+        if needs_artifacts and variant is None:
+            variants = []
+            if args.srcbuild:
+                for entry in os.scandir(bin_dir):
+                    if entry.is_dir():
+                        variants.append(entry.name)
+            if len(variants) == 0:
+                raise FileNotFoundError("no variants found, did you install a board binary package?")
+            elif len(variants) == 1:
+                variant = variants[0]
+            else:
+                raise ValueError("more than one variant found for selected board, specify -V. "
+                    "Found variants: {}".format(" ".join(sorted(variants))))
+        variant_dir = ''
+        if needs_artifacts:
+            if args.srcbuild:
+                variant_dir = variant
+            else:
+                variant_dir = 'kasli-' + variant
```

Comment:
Do not duplicate this code from artiq_flash; share it.

Reply:
If the artiq_flash functionality is now part of artiq_coremgmt why not deprecate artiq_flash?

Reply:
Because artiq_coremgmt doesn't work if there isn't functional firmware already running.